src/types.rs

```diff
@@ -107,6 +107,17 @@ where
     Ok(answer)
 }
 
+fn from_hex_header<'de, D>(deserializer: D) -> Result<block::BlockHeader, D::Error>
```

Comment:
i might be missing something here, but aren't those types like `BlockHeader` already de-serializable? why did you have to re-implement the serializer manually?

Reply:
It just didn't work out of the box

Reply:
it just doesn't work out of the box for me

Reply:
maybe because BlockHeader is missing FromStr